### PR TITLE
Some code and 64 bit fixes

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/lickey/LicenseManager.cpp [hunk 1]
```diff
@@ -471,7 +506,7 @@ namespace lickey
         }
 
         std::ostringstream dataSection(std::ios::binary);
-        unsigned int fileVersion = VERSION();
+        char fileVersion = VERSION();
```
Comment (akio-tanaka) [inline src/lickey/LicenseManager.cpp:509]: this is question:
why do you change the type of fileVersion into char?

**Review by akio-tanaka — COMMENTED** [review]
i put a question. i think that i may not have enough knowledge about it.

it is ok in the other code.

File: src/lickey/LicenseManager.cpp (hunk 1 above)
Comment (ghost) [reply src/lickey/LicenseManager.cpp:509]: implict conversion (or explicit) where fileVersion usage. 

here https://github.com/akio-tanaka/lickey/blob/69ba8fba88e77e99b1f3ab9d3e2ea141248ffc83/src/lickey/LicenseManager.cpp#L378

Comment (akio-tanaka) [reply src/lickey/LicenseManager.cpp:509]: make sense. thank you.

Comment (ghost) [reply src/lickey/LicenseManager.cpp:509]: Okay. I'm ready to implement next features.

Comment (symanli) [reply src/lickey/LicenseManager.cpp:509]: const int validLen = CalcBase64EncodedSize(32) + 8;
        if(static_cast<size_t>(validLen) > decryptedImplSize)
        {
            LOG(error) << "invalid data section";
            return false;
        }

Decryption failed here。

**Pull request merged**